src/platform/forms-system/src/js/routing.js

```diff
@@ -40,8 +40,10 @@ export function getPreviousPagePath(pageList, data, pathname) {
  * @returns {Boolean}
  */
 export function checkValidPagePath(pageList, data, pathname) {
+  // ignore search parameters for custom pages goToPath
+  const name = (pathname || '').split('?')[0];
   return getActiveExpandedPages(pageList, data).some(
-    page => page.path === pathname,
+    page => page.path === name,
```

Comment:
This was added because we are trying a new list-loop pattern to create array entries, so it doesn't work with the `showPagePerItem` setting, which is set up to work with pre-filled arrays